src/main/kotlin/org/rust/ide/intentions/RsShowRecursiveMacroExpansionIntention.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.intentions
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.parentOfType
+import org.rust.ide.actions.macroExpansion.MacroExpansionViewCreator
+import org.rust.lang.core.psi.RsMacroCall
+
+class RsShowRecursiveMacroExpansionIntention : RsElementBaseIntentionAction<RsMacroCall>() {
+    override fun getText() = "Show recursive macro expansion"
+    override fun getFamilyName() = text
+
+    private val recursiveExpansionViewCreator = MacroExpansionViewCreator(expandRecursively = true)
```

Comment:
Follow the [class layout convention](https://kotlinlang.org/docs/reference/coding-conventions.html#class-layout) (not only here).